### PR TITLE
feat(list-box): add disabled styles for listbox selection component

### PR DESCRIPTION
**Pull request by emyarod**
This PR adds disabled styles to the listbox selection component

#### Changelog

**New**

- disabled styles on hover of listbox selection component

Related:

https://github.com/IBM/carbon-components-react/issues/1596

https://github.com/IBM/carbon-components-react/pull/1597

### CODE REVIEW

File: src/components/list-box/_list-box.scss [hunk 1]
```diff
@@ -219,6 +219,10 @@ $list-box-menu-width: rem(300px);
     outline: none;
   }
 
+  .#{$prefix}--list-box__selection--disabled:hover {
```
Comment (asudoh) [inline src/components/list-box/_list-box.scss]:
```suggestion
  .#{$prefix}--list-box--disabled .#{$prefix}--list-box__selection:hover {
```

**Review by asudoh — COMMENTED** [review]
Thank you for jumping in @emyarod! Wondering if the suggestion helps reduce changes in React side.

**Comment by emyarod** [comment]
thanks for the correction @asudoh! this eliminates additional classes needed on the React side

**Review by joshblack — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
Thanks @emyarod for the updates!

**Pull request merged**